snap/info.go

```diff
@@ -41,6 +41,33 @@ import (
 	"github.com/snapcore/snapd/timeout"
 )
 
+// ContainerPlaceInfo offers all the information about where a container (which
+// can be a snap or a component) and its data are located and exposed in the
+// filesystem.
+type ContainerPlaceInfo interface {
+	// ContName returns the name of the container, which is part of the
+	// backing file (so it is the instance name for snaps).
+	ContName() string
+
+	// ContRevision returns the revision of the container.
+	ContRevision() Revision
+
+	// Filename returns the name of the container with the revision
+	// number, as used on the filesystem.
+	Filename() string
+
+	// MountDir returns the base directory of the container.
+	MountDir() string
+
+	// MountFile returns the path where the container file that is mounted is
+	// installed.
+	MountFile() string
+
+	// BinaryNameGlobs returns base name globs that matches all possible
+	// component binaries.
+	BinaryNameGlobs() []string
```

Comment:
Looking at the context where this is used (wrapper binaries), this can be removed from the interface, although this interface looks a bit like the natural place for this method I'd say. It can be removed though.

Reply:
Removed as well